apps/time/src/widgets/time-table/ui/LectureSearchItem.tsx

```diff
@@ -0,0 +1,167 @@
+import {
+  InputHTMLAttributes,
+  type PropsWithChildren,
+  type ReactNode,
+  useMemo,
+} from 'react';
+
+import { CloseOutline } from '@clab-platforms/icon';
+
+interface SearchRemovableTagProps {
+  value: string;
+  onRemove: () => void;
+}
+
+interface SearchFilterProps<T> {
+  title: string;
+  list: T[];
+  selectedItems: T[];
+  onToggleItem: (item: T) => void;
+}
+
+interface SearchDropdownProps extends PropsWithChildren {
+  title: string;
+  isOpen: boolean;
+  onToggle: () => void;
+  selectedItems: ReactNode;
+}
+
+interface SearchInputProps extends InputHTMLAttributes<HTMLInputElement> {
+  inputValue: string;
+  handleInputChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+}
+
+interface SearchDropdownButtonProps {
+  isSelected: boolean;
+  onClick: () => void;
+  children: ReactNode;
+}
+
+function SearchRemovableTag({ value, onRemove }: SearchRemovableTagProps) {
+  return (
+    <button
+      type="button"
+      className="flex min-w-12 items-center justify-between gap-x-1 rounded-md bg-gray-100 px-1.5 py-0 text-sm text-gray-700 hover:bg-gray-200"
+      onClick={onRemove}
+    >
+      <span>{value}</span>
+      <CloseOutline width={16} height={16} />
+    </button>
+  );
+}
+
+function SearchFilter<T extends string>({
+  title,
+  list,
+  selectedItems,
+  onToggleItem,
+}: SearchFilterProps<T>) {
+  const selectedSet = useMemo(() => new Set(selectedItems), [selectedItems]);
+
+  return (
+    <div className="flex w-fit shrink-0 flex-nowrap items-center gap-4">
+      <h3 className="text-sm font-medium">{title}</h3>
+      <div className="border-time-table-border divide-time-table-border flex flex-nowrap divide-x overflow-hidden rounded-md border">
+        {list.map((item) => (
+          <button
+            key={item}
+            type="button"
+            className={`px-3 py-1 text-sm transition-colors hover:bg-gray-300/20 ${
+              selectedSet.has(item)
+                ? 'bg-time-table-header'
+                : 'bg-white text-gray-700'
+            }`}
```

Comment:
```suggestion
            className={cn('px-3 py-1 text-sm transition-colors hover:bg-gray-300/20', selectedSet.has(item) ? 'bg-time-table-header'  : 'bg-white text-gray-700')}
```

코드 전체적으로 cn 함수 사용해서 조건부 스타일 적용해주시면 될 것 같아요